src/test/scala/org/camunda/feel/impl/builtin/BuiltinConversionFunctionsTest.scala

```diff
@@ -62,6 +63,21 @@ class BuiltinConversionFunctionsTest
       ValDateTime("2012-12-24T23:59:00+01:00"))
   }
 
+  it should "convert (DateTime, Timezone)" in {
+    eval("""date and time(@"2020-07-31T14:27:30@Europe/Berlin", "Z")""") should be(
+      ValDateTime(ZonedDateTime.parse("2020-07-31T12:27:30Z")))
+
+    eval(
+      """date and time(@"2020-07-31T14:27:30@Europe/Berlin", "America/Los_Angeles")""") should be(
+      ValDateTime(
+        ZonedDateTime.parse("2020-07-31T05:27:30-07:00[America/Los_Angeles]")))
+
+    eval(
+      """date and time(@"2020-07-31T14:27:30", "Z")""") should be(
+      ValDateTime(
+        ZonedDateTime.parse("2020-07-31T21:27:30Z")))
```

Comment:
:x: The expectation is wrong. 

If a date-time value without a timezone is passed as a function argument then the function should return the given date-time value with the given timezone. There is no calculation needed. It just takes the timezone and adds it to the date-time value.

```suggestion
    eval(
      """date and time(@"2020-07-31T14:27:30", "Z")""") should be(
      ValDateTime(
        ZonedDateTime.parse("2020-07-31T14:27:30Z")))
```